src/pykeen/regularizers.py

```diff
@@ -77,18 +85,29 @@ def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
         """Compute the regularization term for one tensor."""
         raise NotImplementedError
 
-    def update(self, *tensors: torch.FloatTensor) -> None:
+    def update(self, *tensors: torch.FloatTensor) -> bool:
         """Update the regularization term based on passed tensors."""
-        if self.apply_only_once and self.updated:
-            return
+        if not self.training or not torch.is_grad_enabled() or (self.apply_only_once and self.updated):
+            return False
         self.regularization_term = self.regularization_term + sum(self.forward(x=x) for x in tensors)
         self.updated = True
+        return True
```

Comment:
Is this return value used somewhere?

Reply:
Nope, it isn't

Reply:
Should we get rid of it? It introduces a few lines of code which we do not need.

Reply:
Done in 9eefd37f